airflow/configuration.py

```diff
@@ -390,8 +390,14 @@ def _validate_max_tis_per_query(self) -> None:
 
         """
         # TODO: this type of checking can be generalized like the deprecate checking
-        max_tis_per_query = self.getint("scheduler", "max_tis_per_query")
-        parallelism = self.getint("core", "parallelism")
+        try:
+            max_tis_per_query = self.getint("scheduler", "max_tis_per_query")
+            parallelism = self.getint("core", "parallelism")
+        except AirflowConfigException:
+            # in case any of the above config not found, return directly
+            # this could happens in some unit tests
```

Comment:
Which unit tests? We can just add them to the config for them.

Reply:
Here are the tests failed with this config key not found:
```
FAILED tests/core/test_configuration.py::TestConf::test_auth_backends_adds_session - airflow.exceptions.AirflowConfigException: section/key [scheduler/max_tis_per_query] not found in config
FAILED tests/core/test_configuration.py::TestDeprecatedConf::test_deprecated_values_from_conf - airflow.exceptions.AirflowConfigException: section/key [scheduler/max_tis_per_query] not found in config
FAILED tests/core/test_configuration.py::TestDeprecatedConf::test_deprecated_env_vars_upgraded_and_removed[old0-new0] - airflow.exceptions.AirflowConfigException: section/key [scheduler/max_tis_per_query] not found in config
FAILED tests/core/test_configuration.py::TestDeprecatedConf::test_deprecated_env_vars_upgraded_and_removed[old1-new1] - airflow.exceptions.AirflowConfigException: section/key [scheduler/max_tis_per_query] not found in config
FAILED tests/core/test_configuration.py::TestDeprecatedConf::test_deprecated_values_from_environ[conf_dict0] - airflow.exceptions.AirflowConfigException: section/key [scheduler/max_tis_per_query] not found in config
FAILED tests/core/test_configuration.py::TestDeprecatedConf::test_deprecated_values_from_environ[conf_dict1] - airflow.exceptions.AirflowConfigException: section/key [scheduler/max_tis_per_query] not found in config
FAILED tests/core/test_configuration.py::TestDeprecatedConf::test_deprecated_sections[old_config] - airflow.exceptions.AirflowConfigException: section/key [scheduler/max_tis_per_query] not found in config
FAILED tests/core/test_configuration.py::TestDeprecatedConf::test_deprecated_sections[old_config_old_env] - airflow.exceptions.AirflowConfigException: section/key [scheduler/max_tis_per_query] not found in config
FAILED tests/core/test_configuration.py::TestDeprecatedConf::test_deprecated_sections[old_env] - airflow.exceptions.AirflowConfigException: section/key [scheduler/max_tis_per_query] not found in config
FAILED tests/core/test_configuration.py::TestDeprecatedConf::test_deprecated_sections[new_config_old_env] - airflow.exceptions.AirflowConfigException: section/key [scheduler/max_tis_per_query] not found in config
```

These test cases are not using; `conf_vars` to initializing the conf. Instead, they initialize conf from a string directly, like `test_conf = AirflowConfigParser(default_config="")`. I believe it has been done for purpose.

Reply:
Can we not add the needed config to the string?

Reply:
> Can we not add the needed config to the string?

Adding to their configuration string makes them dependent on this module. 

These are unit tests. I don't think it's appropriate tho introduce this dependence to them.

Reply:
But the implementation those tests cover does depend on the configuration. If you do not want they to depend on it, your implementation to the config needs to be changed.

Reply:
@lyso do you want to make those changes ? Or @uranusjr would you be OK leaving the code as is ?